### PR TITLE
Chromium kiosk mode for attached touchscreen

### PR DESCRIPTION
**Pull request by ardeus-ua**

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @ardeus-ua, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;hassbian-scripts#97).

Once you do that we will be able to review and accept this pull request.

Thanks!

**Comment by Landrash** [comment]
@ardeus-ua Can you merge your 3 Pr's into one? It makes it a tad hard to test as is :smile:

**Comment by ardeus-ua** [comment]
Sorry, this was my first pull. Will check how to do it

**Comment by ludeeus** [comment]
@ardeus-ua you can push the other scripts to your branch with name "patch-1"

**Comment by ardeus-ua** [comment]
updated patch-1 branch. Please review

File: package/opt/hassbian/suites/kiosk.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+function hue-show-short-info {
```
Comment (ludeeus) [inline package/opt/hassbian/suites/kiosk.sh:3]: Wrong function should be kiosk-show-short-info

File: package/opt/hassbian/suites/kiosk.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    echo "Kiosk mode for attached touchscreen"
+}
+
+function hue-show-long-info {
```
Comment (ludeeus) [inline package/opt/hassbian/suites/kiosk.sh:7]: Wrong function should be kiosk-show-long-info

File: package/opt/hassbian/suites/kiosk.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    echo "Configures Chromium autostart and points to localhost:8123"
+}
+
+function hue-show-copyright-info {
```
Comment (ludeeus) [inline package/opt/hassbian/suites/kiosk.sh:11]: Wrong function should be kiosk-show-copyright-info

File: package/opt/hassbian/suites/kiosk.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    echo "Copyright(c) 2018 Ardeus <https://github.com/ardeus-ua>"
+}
+
+function kiosk-install-package {
+kiosk-show-short-info
+kiosk-show-copyright-info
+
+apt-get update
+apt-get --no-install-recommends install xserver-xorg xserver-xorg-video-fbdev xinit pciutils xinput xfonts-100dpi xfonts-75dpi xfonts-scalable
```
Comment (ludeeus) [inline package/opt/hassbian/suites/kiosk.sh:20]: Missing `-y` flag, script execution hangs.

File: package/opt/hassbian/suites/kiosk.sh [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+apt-get -y --force-yes install screen checkinstall xterm xdotool x11-xserver-utils unclutter chromium-browser x11vnc expect
```
Comment (ludeeus) [inline package/opt/hassbian/suites/kiosk.sh:21]: --force-yes is deprecated, use one of the options starting with --allow instead.

File: package/opt/hassbian/suites/kiosk.sh [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+# Set up x11vnc
+mkdir /home/pi/scripts
+cp /opt/hassbian/suites/files/setX11vncPass /home/pi/scripts/setX11vncPass
+chmod +x /home/pi/scripts/setX11vncPass
+
+sudo -u pi /home/pi/setX11vncPass raspberry
```
Comment (ludeeus) [inline package/opt/hassbian/suites/kiosk.sh:28]: You are copying this to /home/pi/scripts/setX11vncPass
But are running from /home/pi/setX11vncPass
Resulting in "sudo: /home/pi/setX11vncPass: command not found"
Why not have it in the right dir by default?

File: package/opt/hassbian/suites/kiosk.sh [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+if [ ! -f /home/pi/.vnc/passwd ]; then
+  echo "/home/pi/.vnc/passwd was not created. Trying again."
+  sudo -u pi /home/pi/scripts/setX11vncPass raspberry
```
Comment (ludeeus) [inline package/opt/hassbian/suites/kiosk.sh:31]: The passord should be user defined for this.
With fallback to default if user don't supply one, see mosquitto or homebridge scripts for how you can manage this.

File: package/opt/hassbian/suites/kiosk.sh [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+apt-get clean
+apt-get autoremove -y
+
+cp /opt/hassbian/suites/files/kiosk /home/pi/scripts/kiosk
+chmod +x /home/pi/scripts/kiosk
+echo "sudo -u pi startx /home/pi/scripts/kiosk &" >> /etc/rc.local
+
+echo
+echo "Installation done."
```
Comment (ludeeus) [inline package/opt/hassbian/suites/kiosk.sh:47]: Can you implement an validation check for this script?
See #121 for examples

File: package/opt/hassbian/suites/kiosk.sh [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    exit 1
+  fi
+fi
+
+apt-get clean
+apt-get autoremove -y
+
+cp /opt/hassbian/suites/files/kiosk /home/pi/scripts/kiosk
+chmod +x /home/pi/scripts/kiosk
+echo "sudo -u pi startx /home/pi/scripts/kiosk &" >> /etc/rc.local
```
Comment (ludeeus) [inline package/opt/hassbian/suites/kiosk.sh:44]: This is added after the `exit` in /etc/rc.local and will never be executed.

File: package/opt/hassbian/suites/kiosk.sh [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+echo
+echo "Installation done."
+echo
```
Comment (ludeeus) [inline package/opt/hassbian/suites/kiosk.sh:48]: Since this wont take affect until you reboot the device.
This should be mentioned.

**Review by ludeeus — CHANGES_REQUESTED** [review]
I have now tested this on an fresh install of Hassbian, and there are some issues that need to be corrected, I have added some comments to the files.

Once I moved the script, it did try to execute but gave the this error: https://hastebin.com/dibehoxide

You should also create documentation for this.
See <https://github.com/home-assistant/hassbian-scripts/tree/dev/docs> for inspiration.

**Comment by ludeeus** [comment]
@ardeus-ua are you still working on this?

**Comment by ardeus-ua** [comment]
@ludeeus trying to make it work but no results so far.

**Comment by ludeeus** [comment]
@ardeus-ua  any update on this?

**Comment by ludeeus** [comment]
Closing down this now, it can be revisited at a later stage.

**Pull request closed without merging**